testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/mapper/AbstractAddPrincipalDecoderOnlineTest.java

```diff
@@ -0,0 +1,58 @@
+package org.wildfly.extras.creaper.commands.elytron.mapper;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.wildfly.extras.creaper.commands.elytron.AbstractElytronOnlineTest;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+public class AbstractAddPrincipalDecoderOnlineTest extends AbstractElytronOnlineTest {
+
+    protected static final String TEST_CONSTANT_PRINCIPAL_DECODER_NAME = "CreaperTestConstantPrincipalDecoder";
+    protected static final Address TEST_CONSTANT_PRINCIPAL_DECODER_ADDRESS = SUBSYSTEM_ADDRESS
+            .and("constant-principal-decoder", TEST_CONSTANT_PRINCIPAL_DECODER_NAME);
+    protected static final String TEST_CONSTANT_PRINCIPAL_DECODER_NAME2 = "CreaperTestConstantPrincipalDecoder2";
+    protected static final Address TEST_CONSTANT_PRINCIPAL_DECODER_ADDRESS2 = SUBSYSTEM_ADDRESS
+            .and("constant-principal-decoder", TEST_CONSTANT_PRINCIPAL_DECODER_NAME2);
+    protected static final String TEST_DIFFERENT_PRINCIPAL_DECODER_NAME = "CreaperTestConstantPrincipalDecoder3";
+    protected static final Address TEST_DIFFERENT_PRINCIPAL_DECODER_ADDRESS = SUBSYSTEM_ADDRESS
+            .and("constant-principal-decoder", TEST_DIFFERENT_PRINCIPAL_DECODER_NAME);
+
+    protected static final List<String> PRINCIPAL_DECODERS_1_AND_2
+            = Arrays.asList(TEST_CONSTANT_PRINCIPAL_DECODER_NAME, TEST_CONSTANT_PRINCIPAL_DECODER_NAME2);
+    protected static final List<String> PRINCIPAL_DECODERS_2_AND_DIFFERENT
+            = Arrays.asList(TEST_CONSTANT_PRINCIPAL_DECODER_NAME2, TEST_DIFFERENT_PRINCIPAL_DECODER_NAME);
+
+    private static OnlineManagementClient localClient;
+    private static Operations localOps;
+
+    @BeforeClass
+    public static void addPrincipalDecoders() throws Exception {
+        localClient = createManagementClient();
```

Comment:
IMO this is not nice at all. However, method annotated with @BeforeClass needs to be static.. And it makes much more sense to call this method  only once (per test case) and not before each test.
I think it would be reasonable to return `client`, `ops` and `administration` fields of `AbstractElytronOnlineTest` back to static.